pkg/ctl/upgrade/cluster.go

```diff
@@ -82,31 +84,39 @@ func DoUpgradeCluster(cmd *cmdutils.Cmd) error {
 	if cmd.ClusterConfigFile != "" {
 		logger.Warning("NOTE: cluster VPC (subnets, routing & NAT Gateway) configuration changes are not yet implemented")
 	}
+
 	currentVersion := ctl.ControlPlaneVersion()
 	versionUpdateRequired, err := requiresVersionUpgrade(cfg.Metadata, currentVersion)
 	if err != nil {
 		return err
 	}
 
+	stackManager := ctl.NewStackManager(cfg)
+	stacks, err := stackManager.DescribeStacks()
+	if err != nil {
+		return err
+	}
+
+	if manager.IsClusterStack(stacks) {
+		return upgradeEKSCTLCluster(versionUpdateRequired, cmd.Plan, cfg, currentVersion, ctl, stackManager)
```

Comment:
Should we start a `Cluster interface` that provides `Upgrade` and have two concrete types implement it? Something we can expand as more non-eksctl created operations become possible.

Reply:
Great idea, I'll incorporate it into the PR